### ISSUE TITLE
Questions about action-conditioned Cosmos

### ISSUE THREAD

**Issue by adaGrad1** [NVIDIA/Cosmos#134]
Hi, Cosmos developers! I understand the action-conditioned training code is still yet to be released. While that is still in the works, I was wondering if you could share a bit more information about how the action-conditioned Cosmos models are trained.

1. I understand the action-conditioned models are fine-tuned from Video2World models. Do they incorporate cross-attention with a text prompt? If so, is the text prompt fixed or determined by the prompt upscaler?
2. Can you share more details about how the action was represented in the action-conditioned Bridge dataset experiments in the Cosmos paper? Were actions represented as a delta in gripper position or absolute position target?
3. Can you share more about the embedding MLP which is used before adding the actions to the cross-attention keys/values (for autoregressive) or the timestep embedding (for diffusion)? In other words, how many layers / what input/hidden/output dimension / what activation function?
4. In the autoregressive case, I would assume that each action timestep corresponds to one token in the cross-attention. Is this correct? Is the sampling rate of actions the same as that of the temporally-downsampled frames, or does each action token represent a sample from a higher-frequency polling?
5. Is there any causal masking to ensure the action at timestep t+2 does not affect the prediction at timestep t+1?
6. More broadly, are there any early code snippets you could share elucidating the structure of the ActionCond models? 

Thank you so much! Your work is truly impressive and very exciting to follow.

**Comment by sophiahhuang** [comment]
Thanks for your interest in Cosmos. Please find more details on the action control models in the [paper](https://arxiv.org/abs/2501.03575). We will be releasing [post-training soon](https://github.com/NVIDIA/Cosmos/blob/main/cosmos1/models/POST_TRAINING.md).

**Comment by xavhl** [comment]
Hi is there an eta of the release? 

> Thanks for your interest in Cosmos. Please find more details on the action control models in the [paper](https://arxiv.org/abs/2501.03575). We will be releasing [post-training soon](https://github.com/NVIDIA/Cosmos/blob/main/cosmos1/models/POST_TRAINING.md).

**Comment by xavhl** [comment]
> ...
> 2. Can you share more details about how the action was represented in the action-conditioned Bridge dataset experiments in the Cosmos paper? Were actions represented as a delta in gripper position or absolute position target?
> ...

FYR, action definition was stated on p.46 of the report
<img width="1182" alt="Image" src="https://github.com/user-attachments/assets/2bd0c893-8549-4f75-81bd-a404986e0bd5" />

**Comment by OrangeSodahub** [comment]
@sophiahhuang Hi, could you release the action-conditioned model right now???
